src/snackbar/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { WidgetBase } from '@dojo/framework/widget-core/WidgetBase';
+import { DNode, RenderResult } from '@dojo/framework/widget-core/interfaces';
+import { theme, ThemedMixin } from '@dojo/framework/widget-core/mixins/Themed';
+import { customElement } from '@dojo/framework/widget-core/decorators/customElement';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import * as css from '../theme/snackbar.m.css';
+
+export interface SnackbarProperties {
+	open: boolean;
+	message: string;
+	actionsRenderer?: () => RenderResult;
+	type?: 'success' | 'error';
+	leading?: boolean;
+	stacked?: boolean;
+}
+
+@theme(css)
+@customElement<SnackbarProperties>({
+	tag: 'dojo-snackbar',
+	properties: ['actionsRenderer', 'leading', 'open', 'stacked'],
+	attributes: ['message', 'type']
+})
+export class Snackbar extends ThemedMixin(WidgetBase)<SnackbarProperties> {
+	protected render(): DNode {
+		const { type, open, leading, stacked, message, actionsRenderer } = this.properties;
+		const classes = [css.root];
```

Comment:
Is there a reason you didn't use ternary for adding conditional classes to the root here like we do elsewhere.

Reply:
I changed this to be consistent with what we've done elsewhere. I originally did it this way because I'm not a fan of testing the placement of `null`.

```tsx
const successTemplate = template.setProperty('@root', 'classes', [
	css.root,
	css.open,
	null,
	null,
	css.stacked
]);
```